File: model/src/main/java/net/contrapt/auction/model/WinningBidRepository.java
package net.contrapt.auction.model;

import org.springframework.data.repository.CrudRepository;

import java.util.Collection;
import java.util.List;

/**
 * Created by msimmons on 10/14/14.
 */
public interface WinningBidRepository extends CrudRepository<WinningBid, Long> {

    public List<WinningBid> findByItemIdAndBidderId(Long itemId, Long bidderId);

    public List<WinningBid> findByItemId(Long itemId);

    public List<WinningBid> findByBidderId(Long bidderId);
}

File: model/src/main/kotlin/net/contrapt/auction/model/WinningBidRepository.kt
package net.contrapt.auction.model

import org.springframework.data.repository.CrudRepository

/**
 * Created by msimmons on 10/14/14.
 */
interface WinningBidRepository : CrudRepository<WinningBid, Long> {

    fun findByItemIdAndBidderId(itemId: Long?, bidderId: Long?): List<WinningBid>

    fun findByItemId(itemId: Long?): List<WinningBid>

    fun findByBidderId(bidderId: Long?): List<WinningBid>
}
